PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/LoadJsonConfigFile.java

```diff
@@ -48,16 +64,118 @@ private void loadJsonObject() throws IOException {
             try (BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(inputStream))) {
                 jsonObject = new JSONObject(bufferedReader.lines().collect(Collectors.joining()));
             }
+        } catch (FileNotFoundException e) {
+            LOGGER.log(Level.SEVERE, "Configuration file not found at path: " + CONFIG_FILE_PATH
+                    + ". Please ensure the file exists and is accessible. Error: " + e.getMessage(), e);
+        } catch (JsonParseException e) {
+            LOGGER.log(Level.SEVERE, "Configuration file is not a valid JSON at path: " + CONFIG_FILE_PATH
+                    + ". Please check the file format. Error: " + e.getMessage(), e);
+        } catch (IOException e) {
+            LOGGER.log(Level.SEVERE, "IO error while reading configuration file from path: " + CONFIG_FILE_PATH
+                    + ". Please check file permissions and retry. Error: " + e.getMessage(), e);
         } catch (Exception e) {
-            // Log the error message
-            System.err.println("Error loading JSON configuration: " + e.getMessage());
-            // Rethrow the exception
-            throw e;
+            LOGGER.log(Level.SEVERE, "Unexpected error while loading configuration. " + "Please investigate further. Error: " + e.getMessage(), e);
         }
+
     }
 
-    // The getJsonObject method returns the loaded JSON object
+    /**
+     * The getJsonObject method returns the loaded JSON object
+     * @return JSONObject loaded JSON object
+     */
     public JSONObject getJsonObject() {
         return jsonObject;
     }
+
+
+    /** Method to get the value for a given rule name and key as an Object
+     * If the rule name exists and has the key, return the value
+     * @param ruleName String rule name
+     * @param key String key
+     * @return Object value for the key
+     */
+    private Object getConfigValue(String ruleName, String key) {
```

Comment:
We should not be using Object return type here. This adds for more code to interpret the result.

Reply:
changed it so this is a base method only for other methods returning Strings

Reply:
We should rather return this to be Map<String, Strng> return type that way you can access the json key elements directly with getKey("keyName")